### ISSUE TITLE
[Windows / All?] SIMD headers appear to have several problems

### ISSUE THREAD

**Issue by zshoals** [Kode/Kore#762]
Tried my best here, this stuff is new to me and the #define target specific spam is a bit hard to follow, my bad if I got something wrong.

* Windows 64 bit programs aren't getting the KINC_SSE2 define set because of this line: https://github.com/Kode/Kinc/blob/f6fcbdd3628e5d9d8c6d2d36e6c6d365911ffb0a/Sources/kinc/simd/types.h#L22  _M_IX86_FP == 2 only checks if SSE2 is available for Windows 32 bit programs and is not set if the program is 64 bit. Internet says that SSE2 is always available for Windows 64 bit programs, so at a glance it looks like this needs a (KORE_WINDOWS && KORE_64) check or something.
* The int_x_N headers all need <emmintrin.h> (that's the SSE2 stuff) but only <xmmintrin.h> (SSE) is included. However, I saw that there were tests for the int headers already so I'm not sure if this is just working through some other means? The tests may also themselves be problematic.
* I get the impression from what I've learned so far that \_mm_load_[type] and \_mm_store_[type] are preferable operations compared to \_mm_set_[type] and then extracting values one by one through a union. This might not actually be a problem and just a design/portability decision, but I was wondering about why this was the case.
* All SIMD headers have kinc_[type]xN_load which is calling \_mm_set_[type], and not \_mm_load_[type], which I'd say is misleading.
* kinc_[type]xN_load is using an inconsistent interface between the float header and all other int headers, we should pick one or the other.

Also, this is more of a feature, but I thought I'd mention it here anyway. Even if there isn't any additional functionality added to the float/int headers it might be nice to add more defines for SSE3, 4.1, 4.2, etc. similar to KINC_SSE and KINC_SSE2 that indicate that the platform has these capabilities.

**Comment by RobDangerous** [comment]
Have to have a closer look at the set/store/load-situation before I can say more about that, there indeed might have been portability-issues here - but for everything else, PRs please 🤠

**Comment by zshoals** [comment]
Most of this has been addressed and the naming stuff is not really pressing or important so I'll close this issue.

**Comment by RobDangerous** [comment]
Allrighty. Thanks again for the help!
